packages/flutter_tools/lib/src/build_system/targets/localizations.dart

```diff
@@ -0,0 +1,281 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+import 'package:process/process.dart';
+import 'package:yaml/yaml.dart';
+
+import '../../artifacts.dart';
+import '../../base/file_system.dart';
+import '../../base/io.dart';
+import '../../base/logger.dart';
+import '../../globals.dart' as globals;
+import '../build_system.dart';
+import '../depfile.dart';
+
+/// Run the localizations generation script with the configuration [options].
+Future<Depfile> generateLocalizations({
+  @required LocalizationOptions options,
+  @required String flutterRoot,
+  @required FileSystem fileSystem,
+  @required ProcessManager processManager,
+  @required Logger logger,
+  @required Directory projectDir,
+  @required String dartBinaryPath,
+}) async {
+  // Setup project inputs and outputs. This currently makes some
+  // guess about the output and does not include all files.
+  final Directory inputArb = fileSystem.directory(
+    options.arbDirectory ?? projectDir
+      .childDirectory('lib')
+      .childFile('l10n').uri,
+  );
+  final File outputLocalizations = fileSystem.file(
+    options.outputLocalizationsFile ?? projectDir
+    .childDirectory('lib')
+    .childFile('app_localizations.dart').uri,
+  );
+
+  if (!inputArb.existsSync()) {
+    logger.printError('arb-dir: ${inputArb.path} does not exist.');
+    throw Exception();
+  }
+
+  final List<File> inputs = <File>[
+    if (options.headerFile != null)
+      fileSystem.file(options.headerFile).absolute,
+    // Include all arb files as build inputs.
+    for (final File file in inputArb.listSync().whereType<File>())
+      if (fileSystem.path.extension(file.path) == '.arb')
+        file,
+  ];
+  final List<File> outputs = <File>[
+    if (options.untranslatedMessagesFile != null)
+      fileSystem.file(options.untranslatedMessagesFile).absolute,
+    outputLocalizations,
+  ];
+
+  final Depfile depfile = Depfile(inputs, outputs);
+  final String genL10nPath = fileSystem.path.join(
+    flutterRoot,
+    'dev',
+    'tools',
+    'localization',
+    'bin',
+    'gen_l10n.dart',
+  );
+  final ProcessResult result = await processManager.run(<String>[
+    dartBinaryPath,
+    genL10nPath,
+    if (options.arbDirectory != null)
+      '--arb-dir=${options.arbDirectory.toFilePath()}',
```

Comment:
Add:
```dart
    if (options.outputDirectory != null)
      '--output-dir=${options.outputDirectory.toFilePath()}',
```

Reply:
When this is updated to use a synthetic package the output directory won't be customizable. The tool will need to specify it in a known location so that when we generate the package_config file it points to the right place, and can be picked up by the analyzer and  IDEs